catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -309,6 +317,30 @@ private void handleSuggestionResponse(
     }
   }
 
+  private void handlePartialResults(
+      QueryResponse solrResponse, Map<String, Serializable> responseProps) {
+    if (solrResponse.getResponseHeader() != null) {
+      Boolean partialResults = (Boolean) solrResponse.getResponseHeader().get("partialResults");
+      if (Boolean.TRUE.equals(partialResults)) {
+        responseProps.put("partial-results", true);
+        if (LOGGER.isDebugEnabled()) {
+          String q = "unknown";
+          if (solrResponse.getResponseHeader().get("params") != null
+              && solrResponse.getResponseHeader().get("params") instanceof SimpleOrderedMap) {
```

Comment:
❓ Is it not enough to simply check for `instanceof Map` instead?

Reply:
It does not appear `Map` is actually in the type hierarchy of `SimpleOrderedMap`.  You can see some of the differences from `Map` in the `NamedList` JavaDoc.

https://github.com/apache/lucene-solr/blob/a70a47d053954b0f13b7da6f5612432a8ea7a898/solr/solrj/src/java/org/apache/solr/common/util/NamedList.java#L46-L54